kroxylicious-kms-provider-fortanix-dsm/src/main/java/io/kroxylicious/kms/provider/fortanix/dsm/FortanixDsmKmsEdek.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.fortanix.dsm;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A Fortanix DSM Encrypted Dek.
+ *
+ * @param kekRef - kek reference.
+ * @param edek - edek bytes
+ * @param iv iv bytes
+ */
+record FortanixDsmKmsEdek(@NonNull String kekRef,
+                          @NonNull byte[] edek,
+                          @NonNull byte[] iv) {
```

Comment:
Perhaps swap the order of these two, so it matches the order of the serialized form?